### PR TITLE
Add a skip mechanism for failures that may caused by infra flake

### PR DESCRIPTION
**Pull request by lidizheng**
As title.

b/183904821 secondary_locality_gets_requests_on_primary_failure

```
2021-03-29 02:23:45,958: ERROR    Test case secondary_locality_gets_requests_on_primary_failure failed
Traceback (most recent call last):
  File "tools/run_tests/run_xds_tests.py", line 2045, in <module>
    secondary_zone_instance_group)
  File "tools/run_tests/run_xds_tests.py", line 726, in test_secondary_locality_gets_requests_on_primary_failure
    raise e
  File "tools/run_tests/run_xds_tests.py", line 710, in test_secondary_locality_gets_requests_on_primary_failure
    _WAIT_FOR_BACKEND_SEC)
  File "tools/run_tests/run_xds_tests.py", line 344, in wait_until_all_rpcs_go_to_given_backends
    allow_failures=False)
  File "tools/run_tests/run_xds_tests.py", line 326, in _verify_rpcs_to_given_backends
    raise RpcDistributionError(error_msg)
RpcDistributionError: Backend test-ig-secondary-zone1616980373-nm50 did not receive load
```

Most likely an infra error.

We want to send load to ['test-ig-secondary-zone1616980373-nm50', 'test-ig-secondary-zone1616980373-tgzc'], but only got peer `k02f` and `md9w`. This is completely off, meaning the control plane is feeding out-of-date endpoints. If the propagation isn't completed within 1200s, I don't see there is any point waiting longer.

In cases like this, the test result is meaningless. If this pattern looks good, I can apply it to many other

CC @srini100

---
This change is [Reviewable](https://reviewable.io/reviews/grpc/grpc/25836)

### CODE REVIEW

**Comment by lidizheng** [comment]
CC @sergiitk

File: tools/run_tests/run_xds_tests.py [hunk 1]
```diff
@@ -2728,6 +2756,11 @@ def __init__(self, compute, alpha_compute, project, project_num):
                         client_process.returncode)
                 result.state = 'PASSED'
                 result.returncode = 0
+            except PotentialInfrastructureError as infra_error:
+                logger.exception('Test case %s skipped', test_case)
+                skipped_tests.append(test_case)
+                result.state = 'SKIPPED'
+                result.message = str(e)
```
Comment (sergiitk) [inline tools/run_tests/run_xds_tests.py:2763]: I was trying to figure out from the rendered functions how 'SKIPPED' results are rendered. Unless I'm missing something, the error message is hardcoded:
https://github.com/grpc/grpc/blob/86f93801e5d452cff40d6958d21a8eb211da0654/tools/run_tests/python_utils/report_utils.py#L119-L120

Might make sense to make a fake run and confirm if that hardcoded message `'Skipped'` needs to be replaced with `result.message` when it's not empty.

**Review by sergiitk — APPROVED** [review]
LGTM

**Comment by lidizheng** [comment]
Just discussed with @srini100, we can't mark TD flakes as "SKIPPED", since the xDS interop is testing the entire PSM workflow... So, even if it's not our fault, we still need the dashboard to go red (instead of making it grey).

**Comment by srini100** [comment]
We have to be careful with this change. First, as @lidizheng mentioned, infra errors should not include control plane issues. Even things like ssh flakiness to test clients/servers cannot be treated as infra issues because it could be due to crashed client/server. Only issues like github connectivity or log upload error etc. are infra issues. Second, if we keep skipping tests with infra errors, we have to somehow know that the rate of skipping is very low. Otherwise, we will be compromising on the testing volume.

**Pull request closed without merging**